### DISCUSSION TITLE
What's the technical reason that socket.io/engine.io needs its own server or wrap it and can't just plug into an express route ?

### DISCUSSION THREAD

**Discussion by bogeeee** [socketio/socket.io#4745] [Q&A]
So, why not offer an express router as part of the API, i.e:
```

const myIO = ioExpressRouter({ /* options */ })

app.use("some/path/",  myIO) // use it just as an express router

myIO.on('connection', socket => {
  socket.on('message', data => { });
  ...
}
```

Technically the handler would be able to stream express's http request and response. Is there another reason that makes it impossible ?
Background: I'm trying to use engine.io as a sub handler in the [restfuncs](https://github.com/bogeeee/restfuncs) library which is an RPC on top of http (or websockets)

**Comment by darrachequesne** [comment]
You can simply use the server returned by `app.listen()`:

```js
const app = express()
app.use("/greeterAPI", restfuncs( new GreeterService() ))
const httpServer = app.listen(3000)

const io = new Server(httpServer);
// ...
```
